xbmc/pvr/addons/PVRClient.cpp

```diff
@@ -447,15 +399,81 @@ bool CPVRClient::GetAddonProperties()
   if (retVal == PVR_ERROR_NOT_IMPLEMENTED)
     retVal = PVR_ERROR_NO_ERROR; // timer support is optional.
 
+  /* update the members */
+  std::unique_lock<CCriticalSection> lock(m_critSection);
+  m_clientCapabilities = addonCapabilities;
+  m_timertypes = timerTypes;
+
+  return retVal == PVR_ERROR_NO_ERROR;
+}
+
+bool CPVRClient::GetAddonNameStringProperties()
+{
+  char strBackendName[PVR_ADDON_NAME_STRING_LENGTH] = {};
+  char strConnectionString[PVR_ADDON_NAME_STRING_LENGTH] = {};
+  char strBackendVersion[PVR_ADDON_NAME_STRING_LENGTH] = {};
+  char strBackendHostname[PVR_ADDON_NAME_STRING_LENGTH] = {};
+  std::string strFriendlyName;
+
+  /* get the name of the backend */
+  PVR_ERROR retVal = DoAddonCall(
+      __func__,
+      [&strBackendName](const AddonInstance* addon) {
+        return addon->toAddon->GetBackendName(addon, strBackendName, sizeof(strBackendName));
+      },
+      true, false);
+
+  if (retVal != PVR_ERROR_NO_ERROR)
+    return false;
+
+  /* get the connection string */
+  retVal = DoAddonCall(
+      __func__,
+      [&strConnectionString](const AddonInstance* addon) {
+        return addon->toAddon->GetConnectionString(addon, strConnectionString,
+                                                   sizeof(strConnectionString));
+      },
+      true, false);
+
+  if (retVal != PVR_ERROR_NO_ERROR && retVal != PVR_ERROR_NOT_IMPLEMENTED)
+    return false;
+
+  /* display name = backend name:connection string */
+  strFriendlyName = strlen(strConnectionString) == 0 // connection string is optional
+                        ? strBackendName
+                        : StringUtils::Format("{}:{}", strBackendName, strConnectionString);
```

Comment:
As an idea how to solve the mutex protection problem (and maybe also the localization problem), get rid of the `m_strFriendlyName` member and extend `GetFriendlyName` to calculate the value on demand, from backend name and what not.

Reply:
I will look into that change, that makes the most sense to me

Reply:
Ok changed.  It still doesn't use a mutex but the other Get name functions don't either.